code-name-taran/src/com/spaghettisoft/component/chess/gamelogic/GameLogic.java

```diff
@@ -0,0 +1,906 @@
+		int[] directionsMoveBetweenKingAndMovedFigure = getDirection(moveBetweenKingAndMovedFigure);
+		int directionRowMoveBetweenKingAndMovedFigure = directionsMoveBetweenKingAndMovedFigure[ZERO];
+		int directionColumMoveBetweenKingAndMovedFigure= directionsMoveBetweenKingAndMovedFigure[ONE];
+		
+		if ( (directionRowCurrentMove == directionRowMoveBetweenKingAndMovedFigure) && (directionColumCurrentMove == directionColumMoveBetweenKingAndMovedFigure) ||
+			 (directionRowCurrentMove * MINUS_ONE == directionRowMoveBetweenKingAndMovedFigure) && (directionColumCurrentMove * MINUS_ONE == directionColumMoveBetweenKingAndMovedFigure) ) {
+			return false;
+		}
+		
+		 boolean isSameRow = moveBetweenKingAndMovedFigure.getFromRow() == moveBetweenKingAndMovedFigure.getToRow();
+		 boolean isSameColumn = moveBetweenKingAndMovedFigure.getFromColumn() == moveBetweenKingAndMovedFigure.getToColumn();
+		 boolean isSameDiagonal = Math.abs(moveBetweenKingAndMovedFigure.getSubtractionbetweenRows()) == Math.abs(moveBetweenKingAndMovedFigure.getSubtractionbetweenColums());
+		if ( player.getColor().equals(movedFigure.getColor()) && (isSameRow || isSameColumn || isSameDiagonal) ) {
+			int currentRow = moveBetweenKingAndMovedFigure.getToRow();
+			int currentColum = moveBetweenKingAndMovedFigure.getToColumn();
+			do {
+				currentRow += directionRowMoveBetweenKingAndMovedFigure;
+				currentColum += directionColumMoveBetweenKingAndMovedFigure;
+				if (!IsRowAndColumAreMatrixRange(currentRow, currentColum)) {
+					break;
+				}
+				figure = this.chessBoard.getFigure(currentRow, currentColum);
+				if ( (figure != null) && !figure.equals(movedFigure) ) {
+					if (figure.getColor().equals(player.getColor())) {
+					return false;
+					}
+					result = figure.validMove(new Move(currentRow, currentColum, moveBetweenKingAndMovedFigure.getFromRow(), moveBetweenKingAndMovedFigure.getFromColumn()));
+					break;
+				}
+				
+			} while(true);
+			
+		}
+		return result;
+	}
+	
+	private int[] getDirection(Move move) {
+		int[] directions = new int[TWO];
+		int row = move.getSubtractionbetweenRows();
+		int colum = move.getSubtractionbetweenColums();
+		directions[ZERO] = (int) Math.signum(row) * MINUS_ONE;
+		directions[ONE] = (int) Math.signum(colum) * MINUS_ONE;
+		return directions;
+	}
+
+	private int[] getCurrentPlayerKingPosition(Player player) {
+		int[] position = new int[TWO];
+		for (int i = 0; i < chessBoard.getBoardLenght(); i++) {
+			for (int j = 0; j < chessBoard.getBoardLenght(); j++) {
+				if(isThisFigureKing(i, j) && !isEnemyFigure(player, i, j)) {
+					position[ZERO] = i;
+					position[ONE] = j;
+					return position;
+				}
+			}
+		}
+		return position;
+	}
+	
+	private boolean isThisFigureKnight(int row, int column) {
+		AbstractFigure figure = this.chessBoard.getFigure(row, column);
+		boolean result = false;
+		 result = figure instanceof Knight;
+		return result;
+	}
+	private boolean isThisFigureKnight(AbstractFigure figure) {

```

Comment:
Good Idea to abstract "instance of". The name of the method can be reduced to "isKnight"(ThisFigure is unnecessary).